python/MyriaPythonWorker.py

```diff
@@ -0,0 +1,218 @@
+        elif output_type == DataType.BLOB:
+            cls.write_int(DataType.BLOB, stream)
+            cls.pickle_and_write(obj, stream)
+
+    @classmethod
+    def read_command(cls, stream):
+        length = cls.read_int(stream)
+
+        if length < 0:
+            raise ValueError("Command cannot be less than zero.")
+        s = stream.read(length)
+
+        if len(s) < length:
+            raise EOFError
+        unenc = base64.urlsafe_b64decode(s)
+        return cls.loads(unenc)
+
+    @staticmethod
+    def dumps(obj):
+        protocol = 2
+        return cPickle.dumps(obj, protocol)
+
+    @staticmethod
+    def loads(obj):
+        return cPickle.loads(obj)
+
+    @classmethod
+    def pickle_and_write(cls, obj, stream):
+        serialized = cls.dumps(obj)
+
+        if serialized is None:
+            raise ValueError("Serialized value should not be None.")
+        elif len(serialized) > (1 << 31):
+            raise ValueError("Cannot serialize object larger than 2G.")
+
+        cls.write_int(len(serialized), stream)
+        stream.write(serialized)
+
+
+def main(in_file, out_file):
+    pickle_serializer = PickleSerializer()
+    try:
+        func = pickle_serializer.read_command(in_file)
+        tuple_size = pickle_serializer.read_int(in_file)
+        output_type = pickle_serializer.read_int(in_file)
+        is_flatmap = pickle_serializer.read_int(in_file)
+
+        if tuple_size < 1:
+            raise ValueError("Size of tuple should not be less than 1.")
+
+        while True:
+            num_tuples = pickle_serializer.read_int(in_file)
+            if num_tuples == SpecialLengths.END_OF_STREAM:
+                break
+
+            tuple_list = []
+            for j in range(num_tuples):
+                tuple_list.append(pickle_serializer.read_tuple(in_file, tuple_size))
+
+            retval = func(tuple_list)
+            if is_flatmap > 0:
+                count = len(retval)
+                pickle_serializer.write_int(count, out_file)
+                for i in range(count):
+                    pickle_serializer.write_with_length(
+                        retval[i], out_file, output_type)
+            else:
+                pickle_serializer.write_with_length(retval, out_file, output_type)
+
+            out_file.flush()
+
+    except Exception:
+        try:
+            pickle_serializer.write_int(SpecialLengths.PYTHON_EXCEPTION_THROWN, out_file)
+            pickle_serializer.write_with_length(traceback.format_exc().encode(
+                "utf-8"), out_file, DataType.BLOB)
+            print(traceback.format_exc(), file=sys.stderr)
+        except IOError:
+            # JVM closed the socket
+            print("IOError: \n{}".
+                  format(traceback.format_exc()), file=sys.stderr)
+        except Exception:
+            print("Python worker process failed with exception:\n{}".
+                  format(traceback.format_exc()), file=sys.stderr)
+        exit(-1)
```

Comment:
I am already calling main() in a with clause, so we should not need a finally. I have a try because 
I want to log the errors in case of an exception. Changing exit to sys.exit